code/modules/food_and_drinks/food/condiment.dm

```diff
@@ -69,12 +74,10 @@
 			filling.color = mix_color_from_reagents(reagents.reagent_list)
 			add_overlay(filling)
 
-	else
-		icon_state = "emptycondiment"
-		name = "condiment bottle"
-		desc = "Just your average condiment bottle."
-		customsprite = FALSE
-	. = ..()
+	else if(icon_empty)
+		icon_state = icon_empty
```

Comment:
This wouldn't work if the container starts empty would it ? Since the icon_empty is only assigned if reagents exist.

Reply:
If you somehow spawned a salt or pepper shaker with no reagents in it, yeah, it'd have the default icon showing salt or pepper in it. I can fix that by setting icon_empty on both of them. Everything else just shows its default sprite when empty, which is what you want.